reactivesocket-client/src/main/java/io/reactivesocket/client/ClientBuilder.java

```diff
@@ -151,13 +121,19 @@ public ReactiveSocket build() {
         }
 
         ReactiveSocketConnector<T> filterConnector = connector
-            .chain(socket -> new TimeoutSocket(socket, requestTimeout, requestTimeoutUnit, executor))
+            .chain(socket -> {
+                if (requestTimeout > 0) {
```

Comment:
Does this mean that the `ClientBuilder` instance will always be around? May be a better idea to conditionally add `chain`?